packages/legend-query-builder/src/stores/fetch-structure/tds/projection/QueryBuilderProjectionValueSpecificationBuilder.ts

```diff
@@ -128,8 +128,12 @@ const appendResultSetModifier = (
           );
           limit.values = [
             Math.min(
-              resultModifierState.limit ?? Number.MAX_SAFE_INTEGER,
-              options?.overridingLimit ?? Number.MAX_SAFE_INTEGER,
+              resultModifierState.limit
```

Comment:
im concern on how we are doing the overload data here. Could cause us some bugs further down the line. 
We are mixing logic in what a "builder" is suppose to do.
I think we should be editing the input of runQuery function rather than changing this builder function and having a bunch of downstream effects. 
We are causing us to have to change a whole bunch of roundtrip tests due to this with a bunch of comments even in the preview data. 
The builder should not have to know this in my opinion. 
can we move this part to a separate PR.